frontend/scripts/tests/puppeteer/profile-actor.test.js

```diff
@@ -0,0 +1,274 @@
+/* eslint-disable no-console */
+import puppeteer from 'puppeteer';
+
+import mocks from '../mocks';
+
+let page;
+let browser;
+const baseUrl = 'http://0.0.0.0:8081';
+
+const openBrowser = visible =>
+  visible
+    ? {
+        headless: false,
+        slowMo: 80,
+        args: [`--window-size=1920,1080`]
+      }
+    : { args: ['--no-sandbox'] };
+
+beforeAll(async () => {
+  browser = await puppeteer.launch(openBrowser(false));
+  page = await browser.newPage();
+  await page.setRequestInterception(true);
+  page.on('request', interceptedRequest => {
+    const url = interceptedRequest
+      .url()
+      .replace('https:', '')
+      .replace('http:', '');
+
+    if (url in mocks) {
+      console.warn(`URL (${url}) intercepted and response mocked`);
+      setTimeout(
+        () =>
+          interceptedRequest.respond({
+            status: 200,
+            contentType: 'application/json',
+            headers: {
+              'Access-Control-Allow-Origin': '*'
+            },
+            body: JSON.stringify(mocks[url])
+          }),
+        300
+      );
+    } else {
+      interceptedRequest.continue();
+    }
+  });
+});
```

Comment:
I would move this whole method to some kind of helper and name it accordingly. You could have `helpers` folder for dealing with the configuration things for tests

Reply:
it would be much more readable if it'd be something like:
```
beforeAll(... 
  openBrowser();
  openNewPage();
  setUpForTesting();
  ...
```
wdyt?

Reply:
I agree, started refactoring this on the `profile-place` tests.